### PR TITLE
adding the search-suggest sample-app

### PR DESCRIPTION
**Pull request by olemagnusnorum**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: incremental-search/search-suggestions/_config.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    endpoint : "https://localhost:8080"
+    do_feed  : true
+    do_index_removal_before_feed: false
+    feed_endpoints:
+        - url: http://localhost:8080/
+          indexes:
+              - open_index.json
+        - url: http://localhost:8080/
+          indexes:
+              - open_index.json
```
Comment (kkraune) [inline incremental-search/search-suggestions/_config.yml:40]:
```suggestion
        - url: http://localhost:8080/
          indexes:
              - open_index.json
```

File: incremental-search/search-suggestions/src/main/application/deployment.xml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<deployment version="1.0">
```
Comment (kkraune) [inline incremental-search/search-suggestions/src/main/application/deployment.xml:1]: This file can be removed

File: incremental-search/search-suggestions/src/main/java/ai/vespa/example/searchsuggest/QueryDocumentProcessor.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+import java.util.List;
+import java.util.logging.Logger;
+
+public class QueryDocumentProcessor extends DocumentProcessor {
+    private static final Logger logger = Logger.getLogger(QueryDocumentProcessor.class.getName());
+
+    private static final String QUERY_DOCUMENT_TYPE  = "query";
+    private final List<String> blockWords;
+
+    private final AsyncSession asyncSession;
```
Comment (kkraune) [inline incremental-search/search-suggestions/src/main/java/ai/vespa/example/searchsuggest/QueryDocumentProcessor.java:24]:
```suggestion
```

No need for this one and the code using it below

File: incremental-search/search-suggestions/src/main/java/ai/vespa/example/searchsuggest/QueryDocumentProcessor.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+            if (response.isSuccess()) {
+                logger.info("Success!");
+            }
+            logger.info("In handleResponse: " + response.getTextMessage());
+             */
+        }
+    }
+
+    @Inject
+    public QueryDocumentProcessor(DocumentAccess acc, BlocklistConfig config){
```
Comment (kkraune) [inline incremental-search/search-suggestions/src/main/java/ai/vespa/example/searchsuggest/QueryDocumentProcessor.java:40]:
```suggestion
    public QueryDocumentProcessor(BlocklistConfig config){
```

File: incremental-search/search-suggestions/src/main/java/ai/vespa/example/searchsuggest/QueryDocumentProcessor.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        this.blockWords = config.blocklist();
+        this.asyncSession = acc.createAsyncSession(new AsyncParameters().setResponseHandler(new RespHandler()));
```
Comment (kkraune) [inline incremental-search/search-suggestions/src/main/java/ai/vespa/example/searchsuggest/QueryDocumentProcessor.java:42]:
```suggestion
```

File: incremental-search/search-suggestions/src/main/java/ai/vespa/example/searchsuggest/QueryDocumentProcessor.java [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    }
+
+    public QueryDocumentProcessor(DocumentAccess acc){
```
Comment (kkraune) [inline incremental-search/search-suggestions/src/main/java/ai/vespa/example/searchsuggest/QueryDocumentProcessor.java:45]:
```suggestion
    public QueryDocumentProcessor(){
```

File: incremental-search/search-suggestions/src/main/java/ai/vespa/example/searchsuggest/QueryDocumentProcessor.java [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+        //default constructor typically used for tests
+        this.blockWords = new ArrayList<>();
+        this.asyncSession = acc.createAsyncSession(new AsyncParameters().setResponseHandler(new RespHandler()));
```
Comment (kkraune) [inline incremental-search/search-suggestions/src/main/java/ai/vespa/example/searchsuggest/QueryDocumentProcessor.java:48]:
```suggestion
```

File: incremental-search/search-suggestions/src/main/java/ai/vespa/example/searchsuggest/QueryDocumentProcessor.java [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+        @Override
+        public void handleResponse(Response response) {
+            /*
+            if (response.isSuccess()) {
+                logger.info("Success!");
+            }
+            logger.info("In handleResponse: " + response.getTextMessage());
+             */
+        }
+    }
```
Comment (kkraune) [inline incremental-search/search-suggestions/src/main/java/ai/vespa/example/searchsuggest/QueryDocumentProcessor.java:37]:
```suggestion
```

File: incremental-search/search-suggestions/src/main/java/ai/vespa/example/searchsuggest/QueryDocumentProcessor.java [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public void deconstruct() {
+        super.deconstruct();
+        asyncSession.destroy();
```
Comment (kkraune) [inline incremental-search/search-suggestions/src/main/java/ai/vespa/example/searchsuggest/QueryDocumentProcessor.java:93]:
```suggestion
```

**Review by kkraune — APPROVED** [review]
Good stuff! You might need to remove some more code to make it compile after removing the non-needed asyncsession - please address comments and merge at will :-)

File: incremental-search/search-suggestions/src/main/application/deployment.xml (hunk 2 above)
Comment (olemagnusnorum) [reply incremental-search/search-suggestions/src/main/application/deployment.xml:1]: will pull the new changes and remove the file

**Pull request merged**